python/ml4ir/applications/ranking/features/feature_fns/string.py

```diff
@@ -158,3 +171,179 @@ def call(self, inputs, training=None):
         query_type_vector = self.categorical_vector_op(query_type, training=training)
 
         return query_type_vector
+
+
+class QueryEmbeddingVector(BaseFeatureLayerOp):
```

Comment:
@mohazahran I encourage you to define it as a feature layer instead of feature fn and then have a wrapper feature fn. This allows the layer to be used in model config as well and is more customizable.

Reply:
Let's have this as a todo perhaps.